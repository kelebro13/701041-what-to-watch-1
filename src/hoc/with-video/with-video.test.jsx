import renderer from "react-test-renderer";
import withVideo from "./with-video";

const MockComponent = (props) => <div>{props.renderVideo()}</div>;
MockComponent.propTypes = {
  renderVideo: PropTypes.func
};

const MockComponentWrapper = withVideo(MockComponent);

it(`render correctly withVideo hoc`, () => {
  const film = {
    previewImage: `img/the-grand-budapest-hotel.jpg`,
    previewVideoLink: `https://some-link`
  };

  const tree = renderer
    .create(<MockComponentWrapper isPlaying={false} film={film}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
