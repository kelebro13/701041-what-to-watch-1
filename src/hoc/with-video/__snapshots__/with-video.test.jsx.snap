// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly withVideo hoc 1`] = `
<div>
  <video
    height="175"
    muted={true}
    poster=""
    width="280"
  />
</div>
`;
